### PR TITLE
doc: add installing section for Connected Home over IP dependencies

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Information on how to install GN build system needed by Connected
Home over IP project were added to the nRF Connect SDK
installation guide.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/gs_installing.rst [hunk 1]
```diff
@@ -219,6 +219,51 @@ Therefore, remember to regularly check for updates:
          pip3 install -U west
 
 
+Installing additional toolchains for Connected Home over IP applications development
```
Comment (Damian-Nordic) [inline doc/nrf/gs_installing.rst:222]: How about "toolchains" -> "tools"?

File: doc/nrf/gs_installing.rst [hunk 2]
```diff
@@ -219,6 +219,51 @@ Therefore, remember to regularly check for updates:
...
+====================================================================================
+
+.. note::
+   If you are not interested in the `Connected Home over IP`_ applications, you can ommit steps presented in this section.
+
+`Connected Home over IP`_ project is using `GN`_ meta-build system to generate files for `Ninja`_ and both tools are necessary in order to build an applications.
```
Comment (Damian-Nordic) [inline doc/nrf/gs_installing.rst:228]: did you mean "in order to build a Connected Home over IP application"?

File: doc/nrf/gs_installing.rst [hunk 3]
```diff
@@ -219,6 +219,51 @@ Therefore, remember to regularly check for updates:
...
+However Ninja is also required tool for Zephyr, so it is assumed that it should be installed with the rest of Zephyr's dependencies and GN is the only missing |NCS| requirement.
+
+.. note::
```
Comment (Damian-Nordic) [inline doc/nrf/gs_installing.rst:231]: Should we move the note right below the previous one ("If you are not interested in...")?

File: doc/nrf/gs_installing.rst [hunk 4]
```diff
@@ -219,6 +219,51 @@ Therefore, remember to regularly check for updates:
...
+Download GN binary archive by entering the following commands:
+
+.. tabs::
+
+   .. group-tab:: Linux
+
+      .. parsed-literal::
+         :class: highlight
+
+         wget -O gn_archive https://chrome-infra-packages.appspot.com/dl/gn/gn/linux-amd64/+/latest
```
Comment (Damian-Nordic) [inline doc/nrf/gs_installing.rst:243]: The commands will create `gn_archive` and `gn` files in the current directory which is unknown and which a user may not have write permissions to. Should we rather recommend using a known location? Also gn_archive should be deleted once no longer needed. E.g.

```
cd ~
wget -O gn.zip https://chrome-infra-packages.appspot.com/dl/gn/gn/linux-amd64/+/latest
unzip gn.zip gn
rm gn.zip
```

File: doc/nrf/gs_installing.rst (hunk 3 above)
Comment (kkasperczyk-no) [reply doc/nrf/gs_installing.rst:231]: I don't have strong statement on this, as text is quite short, so second note is not hidden somewhere deep inside the section. I can change it, if you find it useful. Would you prefer to have two notes placed one by one, or both information in the same note?

File: doc/nrf/gs_installing.rst (hunk 2 above)
Comment (kkasperczyk-no) [reply doc/nrf/gs_installing.rst:228]: Yes, isn't it clear enough while whole section applies to the Connected Home over IP?

File: doc/nrf/gs_installing.rst (hunk 1 above)
Comment (kkasperczyk-no) [reply doc/nrf/gs_installing.rst:222]: Ok

File: doc/nrf/gs_installing.rst (hunk 3 above)
Comment (Damian-Nordic) [reply doc/nrf/gs_installing.rst:231]: Yes, I think it's better if such notes are at the beginning so that one knows in advance if it's even useful to read the text for him/her.

File: doc/nrf/gs_installing.rst (hunk 2 above)
Comment (Damian-Nordic) [reply doc/nrf/gs_installing.rst:228]: I'm not sure. Probably @greg-fer will have comments on that ;). But please fix the grammar error: "an applications" -> "an application".

**Review by greg-fer — APPROVED** [review]

**Comment by greg-fer** [comment]
Hey @ru-fu , can you have a look and give your perspective? My edits were merged into the original commit. Thanks!

File: doc/nrf/gs_installing.rst [hunk 5]
```diff
@@ -55,6 +55,61 @@ The installation process is different depending on your operating system.
 
       Also see :ref:`zephyr:mac-setup-alts` for additional information.
 
+If you are interested in building `Connected Home over IP`_ applications, install also the `GN`_ meta-build system to generate files for `Ninja`_.
```
Comment (ru-fu) [inline doc/nrf/gs_installing.rst:58]: Maybe reword a bit, since the user probably doesn't care about generating Ninja files, but wants to use the NCS. ;)
```suggestion
If you are interested in building `Connected Home over IP`_ applications, install also the `GN`_ meta-build system to generate the `Ninja`_ files that the |NCS| uses.
```

File: doc/nrf/gs_installing.rst [hunk 6]
```diff
@@ -55,6 +55,61 @@ The installation process is different depending on your operating system.
...
+
+.. tabs::
+
+   .. group-tab:: Windows
+
+      The Connected Home over IP project currenlty supports only Linux and macOS.
```
Comment (ru-fu) [inline doc/nrf/gs_installing.rst:64]:
```suggestion
      The Connected Home over IP project currently supports only Linux and macOS.
```

Comment (ru-fu) [inline doc/nrf/gs_installing.rst:64]: I think we should start by saying that GN is not supported on Windows yet, because that is what the installation is about. And then go on to say that CHIP supports only Linux and macOS.

File: doc/nrf/gs_installing.rst [hunk 7]
```diff
@@ -55,6 +55,61 @@ The installation process is different depending on your operating system.
...
+      To install the GN tool, complete the following steps:
+
+      1. Download the GN binary archive and extract it by using the following commands:
+
+         .. parsed-literal::
+            :class: highlight
+
+            mkdir ${HOME}/gn && cd ${HOME}/gn
+            wget -O gn.zip https://chrome-infra-packages.appspot.com/dl/gn/gn/linux-amd64/+/latest
+            unzip gn.zip gn
```
Comment (ru-fu) [inline doc/nrf/gs_installing.rst:78]: What does the "gn" at the end do? Do you mean "-d gn" - but that would give you $HOME/gn/gn ... ?

File: doc/nrf/gs_installing.rst [hunk 8]
```diff
@@ -55,6 +55,61 @@ The installation process is different depending on your operating system.
...
+            rm gn.zip
+
+      #. Add the location of the GN tool to the system PATH.
+         For example, if you are using ``bashrc``, run the following commands in the directory with extracted executable file:
```
Comment (ru-fu) [inline doc/nrf/gs_installing.rst:82]:
```suggestion
         For example, if you are using ``bash``, run the following commands:
```
(It doesn't make a difference where you run the commands.)

File: doc/nrf/gs_installing.rst [hunk 9]
```diff
@@ -55,6 +55,61 @@ The installation process is different depending on your operating system.
...
+
+         .. parsed-literal::
+            :class: highlight
+
+            echo 'export PATH=${HOME}/gn:"$PATH"' >> ${HOME}/.bashrc
+            source ${HOME}/.bashrc
+
+   .. group-tab:: macOS
+
+      To install the GN tool, complete the following steps:
```
Comment (ru-fu) [inline doc/nrf/gs_installing.rst:92]: Probably the same changes as for Linux, but I'm not familiar with how it works on macOS.

File: doc/nrf/links.txt [hunk 10]
```diff
@@ -514,3 +516,7 @@
 .. _`Baltimore CyberTrust Root certificate`: https://www.digicert.com/CACerts/BaltimoreCyberTrustRoot.crt.pem
 
 .. _`SKY66112-11 page`: https://www.skyworksinc.com/en/Products/Front-end-Modules/SKY66112-11
+
+.. _`GN` : https://gn.googlesource.com/gn/
+
+.. _`Ninja` : https://ninja-build.org/
```
Comment (ru-fu) [inline doc/nrf/links.txt:522]: Ninja should already be in there - need to update the file.

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: doc/nrf/gs_installing.rst (hunk 7 above)
Comment (kkasperczyk-no) [reply doc/nrf/gs_installing.rst]: Yeah, right, I think I've lost `-d` somewhere and that's why I didn't notice the problem. Anyway I removed that option, as in fact `gn.zip` contains only one executable `gn` file, so I think there is not need to create another directory.

File: doc/nrf/gs_installing.rst (hunk 6 above)
Comment (kkasperczyk-no) [reply doc/nrf/gs_installing.rst]: Ok, I guess it's fixed.

**Review by ru-fu — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

File: doc/nrf/gs_installing.rst [hunk 11]
```diff
@@ -55,6 +55,61 @@ The installation process is different depending on your operating system.
 
       Also see :ref:`zephyr:mac-setup-alts` for additional information.
 
+If you are interested in building `Connected Home over IP`_ applications, install also the `GN`_ meta-build system to generate the Ninja files that the |NCS| uses.
+
+.. tabs::
+
+   .. group-tab:: Windows
+
+      The GN, as well as the Connected Home over IP project, do not support Windows yet.
```
[the remark this answers is not in the document]
Comment (kkasperczyk-no) [reply doc/nrf/gs_installing.rst:64]: Right, that was simplified too much. I rephrased it a little. Is it ok now?

[the remark this answers is not in the document]
Comment (ru-fu) [reply doc/nrf/gs_installing.rst]: This I find confusing now.
If GN is actually supported on Windows and the limitation is in CHIP, we should tell users how to install it here and move the information that CHIP is not supported on Windows to the CHIP user guide. Maybe with a note here to prevent users from planning to use CHIP on Windows.

[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply doc/nrf/gs_installing.rst]: But GN is currently only required by CHIP, so if CHIP is not supported on Windows, we shouldn't tell a user to install GN needlessly.

[the remark this answers is not in the document]
Comment (ru-fu) [reply doc/nrf/gs_installing.rst]: I thought the plan was to just have everybody install it?

[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply doc/nrf/gs_installing.rst]: I'm not aware of such plans. :) I thought that's why there's the following condition at the beginning of the section: "If you are interested in building `Connected Home over IP`_ applications". I believe there are currently no other use cases for GN than building CHIP applications.

[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/gs_installing.rst]: > I thought the plan was to just have everybody install it?

This was one of the options proposed with @LuDuda and discussed at the tech writers' call, but I understood the way to go is to have a subsection under required tools. So this is the option I went for.

[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/gs_installing.rst]: However, I wasn't aware GN is supported on Windows... back to drawing board, then.

File: doc/nrf/gs_installing.rst [hunk 12]
```diff
@@ -55,6 +55,61 @@ The installation process is different depending on your operating system.
...
       Also see :ref:`zephyr:mac-setup-alts` for additional information.
 
+If you are interested in building `Connected Home over IP`_ applications, install also the `GN`_ meta-build system to generate the Ninja files that the |NCS| uses.
+
+.. tabs::
+
+   .. group-tab:: Windows
+
+      The usage of GN by |NCS|, as well as the Connected Home over IP project, is not supported on Windows yet.
+      The Connected Home over IP project currently supports only Linux and macOS.
```
Comment (greg-fer) [inline doc/nrf/gs_installing.rst:65]:
```suggestion
      |NCS| does not support the usage of GN on Windows.
      For this reason, the Connected Home over IP project is not yet supported on Windows.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Added proposal for the Windows tab.
Information about missing support for Windows in user guide are requested in #3670 .

**Review by ru-fu — APPROVED** [review]
Approving under the condition that what we add here corresponds to what is added to the Getting Started Assistant/Toolchain Manager.

**Review by greg-fer — APPROVED** [review]
